Add secondary select action buttons to DataTable

gillespi314 opened the pull request:
Enhance `DataTable` and related components to enable multiple buttons for actions on selected rows

- Create new subcomponent `ActionButton` for `DataTable`
- Add `renderPrimarySelectAction` and `renderSecondarySelectActions`
- Add `renderActionButton`
- Refactor `selectActionButtonText` and `onSelectActionClick` as `primarySelectActionButtonText` and `onPrimarySelectActionButtonText`
- Refactor `ManageHostsPage` to conform to above change
- Refactor `TextCell` component to enable it to display `boolean` values in text

TODO 

- Test coverage
- Adjust styling of table headers to fix problems with borders being cut off
<img width="1085" alt="Screen Shot 2021-07-25 at 1 26 44 PM" src="https://user-images.githubusercontent.com/73313222/126909438-b2f0dceb-fff1-4485-b0b0-5c168ce011e0.png">
<img width="98" alt="Screen Shot 2021-07-25 at 1 27 10 PM" src="https://user-images.githubusercontent.com/73313222/126909442-9d7f3f14-e950-4d49-a8c3-d11e04a3e0a5.png">

gillespi314 commented:
@RachelElysia, I split out the proposed `DataTable` changes from #1426 so that you could use them for your work in progress. To see how the new functionality can be incorporated into an existing page, you can check out the changes I made to the `Teams` page (I reverted those changes in this [commit](https://github.com/fleetdm/fleet/pull/1470/commits/71f2c6e210cc08c47be55f89ebd3eb7a810bc4f1) because they were just for testing/demonstration purposes).  Happy to discuss whenever you'd like!

RachelElysia commented:
Definitely will check this out in the AM. I know my wireframes have an icon button in the TableContainer so I'll just double-check that is/can be added. See you tomorrow!

gillespi314 commented:
@RachelElysia, I updated the `ActionButton` component to accept an `iconLink` prop that should enable what your wireframes require.  I also added a `<div>` that includes `classNames` to wrap the `ActionButton` so each icon and link can be styled independently in the page styles.  Check out this [commit](https://github.com/fleetdm/fleet/pull/1470/commits/d16870318086fe597e4985e77feb31d58b95e0fd) if you want to see an illustrative example of how it can all be used together.

noahtalerman reviewed (approved):
Looks good to me!

noahtalerman reviewed (changes requested):
Actually, after a second pass, it looks like there's a bug in which I can't select individual checkboxes in the "Hosts" table. Please wait to merge these changes in. Recording a Loom video to capture the behavior now

noahtalerman reviewed (approved):
Ok, the checkbox issue appears to be unrelated to these changes. I filed a separate bug issue here: https://github.com/fleetdm/fleet/issues/1475
